### PR TITLE
use reverse-iterator if possible

### PR DESCRIPTION
**Pull request by navis**
Addressing missed comments https://github.com/druid-io/druid/pull/2014/files#r49129298

### CODE REVIEW

File: processing/src/main/java/io/druid/segment/BitmapOffset.java [hunk 1]
```diff
@@ -51,8 +52,9 @@ private IntIterator newIterator()
     if (!descending) {
       return bitmapIndex.iterator();
     }
-    // ImmutableRoaringReverseIntIterator is not cloneable.. looks like a bug
-    // update : it's fixed in 0.5.13
+    if (bitmapIndex instanceof WrappedImmutableRoaringBitmap) {
+      return ((WrappedImmutableRoaringBitmap) bitmapIndex).getBitmap().getReverseIntIterator();
+    }
     int i = bitmapIndex.size();
     int[] back = new int[bitmapIndex.size()];
```
Comment (xvrl) [inline processing/src/main/java/io/druid/segment/BitmapOffset.java]: I didn't notice this earlier, but materializing the integer array might create a lot of heap pressure. Instead, why not copy the bitmap into a temporary roaring bitmap and use its reverse iterator?

Comment (himanshug) [inline processing/src/main/java/io/druid/segment/BitmapOffset.java]: we noted this before as well ( https://github.com/druid-io/druid/pull/2014/files#r48668522 ).

Comment (navis) [inline processing/src/main/java/io/druid/segment/BitmapOffset.java]: ok

**Comment by fjy** [comment]
:+1: after squashing commits

**Comment by xvrl** [comment]
:+1: once commits are squashed

**Comment by nishantmonu51** [comment]
:+1:, will merge once travis passes.

**Pull request merged**